src/main/kotlin/com/shankhadeepghoshal/kotlinfinaltask/controller/Controller.kt

```diff
@@ -0,0 +1,29 @@
+package com.shankhadeepghoshal.kotlinfinaltask.controller
+
+import com.shankhadeepghoshal.kotlinfinaltask.pojo.Dog
+import com.shankhadeepghoshal.kotlinfinaltask.pojo.Url
+import com.shankhadeepghoshal.kotlinfinaltask.service.DogsApiService
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import mu.KotlinLogging
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class Controller(private val dogsApiService: DogsApiService) {
+    private val log = KotlinLogging.logger {}
+
+    @GetMapping(path = ["/dogs"], produces = [MediaType.APPLICATION_NDJSON_VALUE])
+    suspend fun getAllDogs(): Flow<Dog> {
+        log.info { "Request incoming for fetching all dogs" }
+        return dogsApiService.findAllDogs()
+    }
+
+    @GetMapping(path = ["/{breedName}/url"], produces = [MediaType.APPLICATION_NDJSON_VALUE])
+    suspend fun getUrlsByMainBreedName(@PathVariable breedName: String): Flow<Url>? {
+        log.info { "Request incoming for fetching URLs of breed: $breedName" }
+        return dogsApiService.findDogsUrl(breedName)?.asFlow()
+    }
+}
```

Comment:
Suggestion on proper api definition. It's best to start with the version of the api, then use resource name. E.g. in your case it could be:
```
/v1/dogs
/v1/dogs/{breedName}/url
```

Reply:
Usually I do it the other way, the resource name and then the API version. So it follows something like -> thingsNotMuchLikelyToChange(resourceName)/thingsThatWillseldomChange(version)/thingsThatWillChangeMoreOften(sub-resource)/thingsThatWillChangeALot(path variables).
In this case I don't really care because I'm not seriously planning to build a production ready app with this piece of code

Reply:
```
Usually I do it the other way, the resource name and then the API version. 
```
Also, an option.
```
In this case I don't really care because I'm not seriously planning to build a production ready app with this piece of code
```
Designing a decent api even in training tasks shows a mature approach to building an api. "Practice makes perfect".